### ISSUE TITLE
Import image stuck darktable

### ISSUE THREAD

**Issue by Nilvus** [darktable-org/darktable#4170]
**Describe the bug**
Clicking on image button in import options (or when dialog window open on a RAW file) makes darktable stuck. Need to force quit darktable. 
~~Reproducible each time on your last commit, @TurboGit, 092f7408a98ed009163e87484bfba9359283bdb2.~~

**To Reproduce**
Just click on import image button. Import folder is ok.

**Expected behavior**
Having dialog window and possibility to import correctly image.

**Platform (please complete the following information):**
darktable master
Ubuntu 19.10

**Comment by Nilvus** [comment]
Forget to wait finishing recompiling again your last commit and now it's working. I made same steps on compiling (even removing build folder and darktable install one before). After having this issue on your commit then after compiling with previous commit where that was working, or to be more precise, I've had the dialog window but it sucks then when selecting image (before clicking on open button).

**Comment by Nilvus** [comment]
Last test made : at first I have had the stuck problem with no dialog import window opening on this button always on last commit cited and not on other before (on tests made tonight) and now after recompiling last one and previous one, I had dialog window opening on two commits cited on start and now have darktable stuck on that window on both commits. So it seems that's not related to 092f7408a98ed009163e87484bfba9359283bdb2 commit. 

Quite strange behavior. Will try to make some more tests and previous commits to better understanding this strange issue...

**Comment by aurelienpierre** [comment]
I have this behaviour too, it's been a couple of days (with e28c49986a03566c01e72e3f4ef15152cc32f217 and already a bit before). When importing a single image, as soon as I click on the picture, darktable hangs forever. It seems only some raw do that, not TIFF. I'm busy on another part of the code, so I didn't investigate more for now.

**Comment by Nilvus** [comment]
@aurelienpierre : thanks for reporting that. More tests I've made since show that was older indeed. I use that option not really often. So a bug to fix (need some labels ;-) !)

**Comment by TurboGit** [comment]
This is due to recent work on metadata.

@phweyland : Can you have a look ? I have seen that dt is locked into _get_tb_added_metadata_string_values().

**Comment by phweyland** [comment]
> @phweyland : Can you have a look ? I have seen that dt is locked into _get_tb_added_metadata_string_values().

Yes, I'll do.

**Comment by TurboGit** [comment]
Issue seems to be in _metadata_execute(imgs, metadata, &undo, undo_on, DT_MA_ADD); 
called from metadata.c:627

That's what I have learnt from initial investigation.

**Comment by phweyland** [comment]
I cannot reproduce this with the current master.
Is there a commit more sensitive than the others ?

**Comment by TurboGit** [comment]
From master I can reproduce, click on import folder, select a folder where you have many picture. select import picture and it may stuck.

**Comment by TurboGit** [comment]
On my side after selecting multiple time "import images" I was able to reproduce.

**Comment by phweyland** [comment]
nop ... 
could you share an image which fails ?

**Comment by phweyland** [comment]
imported > 20.000 images with no issue.
If that's related to metadata change from last week end, it can come from the metadata which are imported, special character, ... that you may have in your images but I haven't.

Faulty images would be welcome to find out the issue. 

On the other hand I will try to find another solution to deal with apostrophe.

**Comment by aurelienpierre** [comment]
The raw from there (https://forums.darktable.fr/showthread.php?tid=4466&pid=38236#pid38236) makes  it crash 100% of times for me.

**Comment by phweyland** [comment]
for that image I get this (but dt opens the image):

```
[exiv2] /home/philippe/Desktop/Images/Photos/2020/2020/20191023_tournai_0010.NEF: /home/philippe/Desktop/Images/Photos/2020/2020/20191023_tournai_0010.NEF: The file contains data of an unknown image type
[rawspeed] (20191023_tournai_0010.NEF) virtual std::unique_ptr<rawspeed::RawDecoder> rawspeed::RawParser::getDecoder(const rawspeed::CameraMetaData*), line 91: No decoder found. Sorry.
allocation failed???
[exiv2] /home/philippe/Desktop/Images/Photos/2020/2020/20191023_tournai_0010.NEF: /home/philippe/Desktop/Images/Photos/2020/2020/20191023_tournai_0010.NEF: The file contains data of an unknown image type
```

**Comment by phweyland** [comment]
Sorry for the previous message. My mistake.
This image opens properly for me (current master).

But I've found a difference between import folder / image. In the second case I get records in meta_data table with imgid = 0. This is not good. I'll try to debug.

On the other hand I've found a simple way to allow apostrophe in metadata.

**Comment by Mark-64** [comment]
#4151 is also metadata related

**Comment by rabauke** [comment]
Similar issues here (git master). Import folder works fine, opening import file dialog freezes darktable immediately. I was able to work around this by resetting `ui_last/import_last_directory` in `darktablerc` to another directory with no or a smaller number of images. The problematic folder contains approx 1500 images. Will test if #4176 fixes this issue for me.

**Comment by Nilvus** [comment]
Fixed by #4176. Thanks.

**Comment by phweyland** [comment]
> I was able to work around this by resetting `ui_last/import_last_directory` in `darktablerc` to another directory with no or a smaller number of images. The problematic folder contains approx 1500 images

That would mean that import image scans the full folder (and gets metadata for each image) before the user selects the image to be imported ? Would that be useful ? I'll check that.
